phoenix-core/src/main/java/org/apache/phoenix/query/KeyRange.java

```diff
@@ -108,6 +109,34 @@ public KeyRange apply(byte[] input) {
         }
     };
 
```

Comment:
@dbwong  ```KeyRange.COMPARATOR.reversed()``` will not give the correct behavior. 

For eg for the query Where COL1='1' OR COL1='12' OR COL1='2'
the key range before sorting is

```
0 = {KeyRange@8864} "\xCE" -> "1"
1 = {KeyRange@8865} "\xCE\xCD" -> "12" 
2 = {KeyRange@8866} "\xCD" -> "2"
```

Sorting using  ```KeyRange.COMPARATOR``` gives the following which isn't correct.
```
0 = {KeyRange@8879} "\xCD" -> "2"
1 = {KeyRange@8880} "\xCE" -> "1"
2 = {KeyRange@8881} "\xCE\xCD" -> "12"
```
Using ```KeyRange.COMPARATOR.reverse()``` just reverses the following order which isn't correct.

We need "12" to sort before "1" as thats how its stored on disk. Using ```DescVarLengthFastByteComparisons``` gives  the correct ordering:
```
0 = {KeyRange@8899} "\xCD" -> "2"
1 = {KeyRange@8900} "\xCE\xCD" -> "12
2 = {KeyRange@8901} "\xCE" -> "1"
```

Reply:
Ok I looked in more detail and I agree.  While this may not be the ideal fix I'm still a bit worried though on if we have a region boundary there could still be issues as we may not intersect properly as the data on disk will contain the FF.  I don't think this should hurt though maybe we should add this note to https://issues.apache.org/jira/browse/PHOENIX-4306 as something we want to test in the future.